controller/Redirector.php

```diff
@@ -57,17 +58,22 @@ public function redirectTaskToInstance()
         $urlBuilder = $this->getServiceLocator()->get(ResourceUrlBuilder::SERVICE_ID);
         $resource = $this->getResource($uri);
 
-        if ($resource->exists()) {
-            return $this->returnJson([
-                'success' => true,
-                'data'    => $urlBuilder->buildUrl($resource)
-            ]);
+        try {
+            $this->returnJson(
+                [
+                    'success' => true,
+                    'data' => $urlBuilder->buildUrl($resource),
+                ]
+            );
+        } catch (Throwable $exception) {
+            $this->returnJson(
+                [
+                    'success' => false,
+                    'errorCode' => 202,
+                    'errorMessage' => __('The requested resource does not exist or has been deleted'),
```

Comment:
This is not the only error (exception that can happen). Check `ResourceUrlBuilder::buildUrl`.

Reply:
The main idea was to avoid double-checking for `$resource->exists()` in 2 places:
1. `controller/Redirector::redirectTaskToInstance`
2. `model/routing/ResourceUrlBuilder::buildUrl`

We do not need to check in `controller/Redirector::redirectTaskToInstance` as we are doing the same at the beginning of `model/routing/ResourceUrlBuilder::buildUrl` and throw an exception.

Reply:
Yes, I got your idea and handling exceptions is always better than `bool` that do not explain the errors. But... we can have different causes of errors and in the UI message we do not explain it or have any logs.

Reply:
I don't want to change old logic so much, but I can add logs for us, so if we will receive any error - we can check logs and fix it easily.

Is it ok?

Reply:
Or I can do the following:
```php
try {
    $this->setSuccessJsonResponse($this->getResourceUrlBuilder()->buildUrl($resource));
} catch (InvalidArgumentException $exception) {
    $this->setErrorJsonResponse(
        __('The requested resource does not exist or has been deleted'),
        202,
        [],
        202
    );
} catch (Throwable $exception) {
    $this->logError($exception->getMessage());
    $this->setErrorJsonResponse(
        __('There was a problem redirecting to the requested resource.'),
        500,
        [],
        500
    );
}
```

What do you prefer?

Reply:
Adding a log for both cases is important IMO :)